engine/src/flutter/lib/web_ui/lib/src/engine/lazy_path.dart

```diff
@@ -396,15 +455,22 @@ class LazyPath implements LayerPath, Collectable {
     double end, {
     bool startWithMoveTo = true,
   }) {
-    return LazyPath._(
-      constructors,
-      pathMetric.iterator.path._fillType,
-      () => pathMetric.buildExtractedPath(start, end, startWithMoveTo: startWithMoveTo),
-    );
+    return LazyPath._(constructors, pathMetric.iterator.path._fillType, () {
+      // The extracted path is only needed temporarily to create the path builder. It can be disposed
+      // immediately after the path builder is created.
+      final DisposablePath path = pathMetric.buildExtractedPath(
+        start,
+        end,
+        startWithMoveTo: startWithMoveTo,
+      );
+      final DisposablePathBuilder pathBuilder = constructors.fromPath(path);
+      path.dispose();
+      return pathBuilder;
+    });
   }
 
   DisposablePathConstructors constructors;
-  DisposablePath Function() initializer;
+  DisposablePathBuilder Function() initializer;
```

Comment:
I think you might be able to just remove this completely, since you've simplified the factory constructors.